### PR TITLE
Swtich to pipenv for Python requirements management

### LINKED ISSUE #612: Build breaks if unlocked transitive dependencies make breaking changes
## overview

We have our dependency versions locked down, but if a downstream dependency is unlocked and makes a breaking change, our build still breaks.

## behavior

See discussion around https://github.com/aio-libs/aiohttp/issues/2662

## steps to reproduce

See https://travis-ci.org/Opentrons/opentrons/builds/329179831?utm_source=github_status&utm_medium=notification

### PR DESCRIPTION
**Pull request by btmorr**
## overview

Pipenv provides a lockfile for both direct and transitive dependencies, as well as easier management of virtual environments. This prevents builds from breaking when an unlocked transitive dependency updates. Also upgrade aiohttp to pick up bugfix.

Fixes #612 

## changelog

- (feature) Add Pipfile and Pipfile.lock and update documentation for use of `pipenv`
- (chore) Update docker container and Makefile to use `pipenv`
- (bugfix) Upgrade aiohttp to v2.3.8 to pick up fix for breaking change in yaml 1.0.0

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,7 @@ PROTOCOL_DESIGNER_DIR := protocol-designer
 #   https://github.com/electron-userland/electron-builder/issues/2222
 .PHONY: install
 install:
+	pip install pipenv
```
Comment (mcous) [inline Makefile]: Will this interfere if the user already has pipenv installed (e.g. via Homebrew)?

File: api/Pipfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+name = "pypi"
+
+
+[packages]
+
+pyserial = "==3.2.1"
+aiohttp = "==2.3.8"
+numpy = "==1.12.1"
+urwid = "==1.3.1"
+"e1839a8" = {path = ".", editable = true}
```
Comment (mcous) [inline api/Pipfile]: Oh weird, does pipenv do some sort of hashing on the local directory? Or is this a commit hash or something?

File: api/README.rst [hunk 3]
```diff
@@ -114,8 +114,45 @@ Installing Opentrons API
 ------------------------
 If you are just starting with Python it is recommended to install Jupyter notebook to run Opentrons API. Please refer to `Full API Documentation`_ for detailed instructions.
 
-If you are familiar with python and comfortable running ``pip``, you can install Opentrons API by running:
+If you are familiar with python and comfortable running ``git`` and ``pip``, first, check out the repository and install `pipenv`:
```
Comment (mcous) [inline api/README.rst]: These are great instructions! It does seem, though, that the existing instructions were for installing `opentrons` from PyPI. Do we want a separate section for dev setup instructions (might need to be synced up with and/or linked to from the repo readme)? Sorry and ignore if I'm just unaware of API distribution / documentation plans

File: Makefile (hunk 1 above)
Comment (OpentronsMfg) [reply Makefile]: Nope, it should just say "Requirement already satisfied" and move on

File: api/Pipfile (hunk 2 above)
Comment (OpentronsMfg) [reply api/Pipfile]: I'll have to look deeper into it. Either way, additional commits do not mess it up, so it's not an immediate risk

File: api/README.rst (hunk 3 above)
Comment (OpentronsMfg) [reply api/README.rst]: I talked to @willcanine about PyPi yesterday. PyPi is not actually designed for distributing applications--it's designed around libraries. If you want to distribute an application, there are some funny things you have to do. Also, if you look in our PyPi, it's on 2.5, so we're not currently publishing there in a meaningful way. Couple all of that with the fact that our app no longer "simply works" on Windows and I'd say that we shouldn't front that in the docs as the primary way of getting and running the server.

File: Makefile (hunk 1 above)
Comment (mcous) [reply Makefile]: Just FYI, if I `brew install pipenv`, this `pip install pipenv` overrides my homebrew pipenv

**Pull request merged**